### ISSUE TITLE
Move AuthRPC client inside lib/auth

### ISSUE THREAD

**Issue by fredmaggiowski** [nexocrew/3nigm4#44]
Let's make the auth library self-contained by letting her export even the RPC Client (and avoid custom/copy-paste implementation in each package)

**Comment by fredmaggiowski** [comment]
@dystonie 

I've refactored auth and db packages and unified all the database mocks in a single one (in `lib/database/mock`. 

Now everything compiles fine with `make` but the following tests in `storageservice` are failing:
- TestStorageResourceFlow;
- TestStorageUploadResourceDuplicated;
- TestStorageDeleteJobDuplicated

I gave a look but I honestly I'm not 100% clear of what is happening there..

**Comment by dystonie** [comment]
@FedericoMaggi 

I do not agree with the aggregation you have done: database frontend are service specific and implements the minimal functions required by the specific associated service, the should be included with the service itself (also for better being able to add functionalities if required). Consider that these interfaces are not logically intended to wrap all used database capabilities.

I agreed with your idea about separating (inside a subfolder in the service dir for ex. "auth/mock") the mock implementations but I do not see the point in dividing actual database wrapping function: it makes less readable code; less maintainable and mix functionalities absolutely non related.

I would suggest restoring the previously organization and only proceed with mock separation.

**Comment by fredmaggiowski** [comment]
To summarise our talk about this issue today:
- lib/auth is OK
- lib/database should be deleted and all the `database.go` and mock files moved to each package.

I'd consider renaming the file from `database.go` to something more evocative of what it really is (since it is _NOT_ a database wrapper)

Mocks will be placed in a _mock_ subpackage hence for example we'll have:

```
3nigm4/<service>/mock
```
